dsalgo/src/site/ds/LinkedList/index.js

```diff
@@ -1,375 +1,386 @@
+              <Element
+                data={{ value: curr.info }}
+                type="LinkedList"
+                next={true}
+                highlight={curr.highlight}
+              />
+            </Fragment>
+          );
+        } else {
+          list.push(
+            <Fragment key={key + "-" + curr.info}>
+              <Element
+                data={{ value: curr.info }}
+                type="LinkedList"
+                next={false}
+                highlight={curr.highlight}
+              />
+            </Fragment>
+          );
         }
-
-        return (
-            <Container>
-                <Row>
-                    <Col sm={3}>
-                        <Insert parent={this}/>
-                    </Col>
-                    <Col sm={4}>
-                        <Delete parent={this}/>
-                    </Col>
-                    <Col sm={3}>
-                        <Update parent={this} />
-                    </Col>
-                    <Col sm={2}>
-                        <Search parent={this} />
-                    </Col>
-                </Row>
-                <Row className="mt-4 mb-4">
-                    {list}
-                </Row>
-            </Container>
-        );
+        curr = curr.next;
+        key++;
+      }
     }
-}
\ No newline at end of file
+    setList(list);
+    setRendered(true);
+  }
+
+  //-----------------content of render function ------------------------------------
+  React.useEffect(() => {
+    if (!rendered)
+      renderList();
+  });
+
+
+
+  return (
+    <div className={gridclass.root} >
+      <Grid
+        container
+        direction="row"
+        justify="flex-end"
+        alignItems="center"
+
+      >
+        <Grid item xs={6} className="mr-5 mt-5" sm={2} spacing={3} >
+          <Paper className={gridclass.paper}>
+            <Card>
+              <h1 className="bg-primary text-center text-white" >Operations</h1>
+              <CardContent className="pl-0 pr-0 pt-0 text-left">
+
+
+                <FormControl className="pl-3" component="fieldset">
+                  <FormLabel component="legend"></FormLabel>
+                  <RadioGroup aria-label="gender" name="customized-radios">
+                    <FormControlLabel value="Insert" onChange={showcomp} control={<StyledRadio />} label="Insert" />
+                    <FormControlLabel value="Delete" onChange={showcomp} control={<StyledRadio />} label="Delete" />
+                    <FormControlLabel value="Update" onChange={showcomp} control={<StyledRadio />} label="Update" />
+                    <FormControlLabel value="search" onChange={showcomp} control={<StyledRadio />} label="Search" />
+
+                  </RadioGroup>
+                </FormControl>
+
+              </CardContent>
+
+            </Card>
+
+
+          </Paper>
+        </Grid>
+
+
+        <Grid className="mr-5 mt-5 pl-4" item spacing={4} xs={6} sm={2}>
+          <Insert
+            show1={compshow}
```

Comment:
show1 is not a good variable name. Change it to showCard. Also, condition inside {} for prop value should evaluate to true or false. For example compShow === 'Insert' should be prop passed as true or false for prop name showCard ( show1  here ).